### PR TITLE
Remove PlainPointer, RelativePointer and RelativeFixupPointer

### LINKED ISSUE #57706: Remove PlainPointer/RelativePointer/RelativeFixupPointer
These can all now be either completely normal pointers or DAC'ized pointers as NGEN is now gone.

### PR DESCRIPTION
**Pull request by davidwrighton**
- Removed all uses of PlainPointer
- Removed RelativePointer usage from vtable, and MethodDesc logic
-  This PR is WIP as the majority of RelativePointer code isn't yet deleted.

Fixes #57706

### CODE REVIEW

File: src/coreclr/vm/field.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ class FieldDesc
 #endif
 
   protected:
-    RelativePointer<PTR_MethodTable> m_pMTOfEnclosingClass;  // This is used to hold the log2 of the field size temporarily during class loading.  Yuck.
+    PTR_MethodTable m_pMTOfEnclosingClass;  // This is used to hold the log2 of the field size temporarily during class loading.  Yuck.
```
Comment (jkoritzinsky) [inline src/coreclr/vm/field.h:42]: In a future cleanup PR, we should make this field a union to make the "Yuck" comment a little more formalized so we aren't just storing an integer in a "dac-ized" pointer field.

File: src/coreclr/vm/jitinterface.cpp [hunk 2]
```diff
@@ -8520,8 +8498,7 @@ void CEEInfo::getMethodVTableOffset (CORINFO_METHOD_HANDLE methodHnd,
 
     *pOffsetOfIndirection = MethodTable::GetVtableOffset() + MethodTable::GetIndexOfVtableIndirection(method->GetSlot()) * TARGET_POINTER_SIZE /* sizeof(MethodTable::VTableIndir_t) */;
     *pOffsetAfterIndirection = MethodTable::GetIndexAfterVtableIndirection(method->GetSlot()) * TARGET_POINTER_SIZE /* sizeof(MethodTable::VTableIndir2_t) */;
-    *isRelative = MethodTable::VTableIndir_t::isRelative ? 1 : 0;
-    _ASSERTE(MethodTable::VTableIndir_t::isRelative == MethodTable::VTableIndir2_t::isRelative);
+    *isRelative = 0;
```
Comment (jkoritzinsky) [inline src/coreclr/vm/jitinterface.cpp:8501]:
```suggestion
    *isRelative = false;
```

**Review by jkoritzinsky — APPROVED** [review]

File: src/coreclr/vm/jitinterface.cpp [hunk 3]
```diff
@@ -3104,12 +3099,6 @@ void CEEInfo::ComputeRuntimeLookupForSharedGenericToken(DictionaryEntryKind entr
                 IfFailThrow(sigptr.GetData(&data));
                 pResult->offsets[2] = sizeof(TypeHandle) * data;
 
-                if (MethodTable::IsPerInstInfoRelative())
-                {
-                    pResult->indirectFirstOffset = 1;
-                    pResult->indirectSecondOffset = 1;
```
Comment (elinor-fung) [inline src/coreclr/vm/jitinterface.cpp:3110]: Are `indirectFirstOffset` and `indirectSecondOffset` always 0 now / could they be removed in the future?

**Review by elinor-fung — APPROVED** [review]

File: src/coreclr/vm/jitinterface.cpp (hunk 3 above)
Comment (davidwrighton) [reply src/coreclr/vm/jitinterface.cpp:3110]: No, they may still be needed for cross-dll handling in NativeAOT in the future. So, its not clear its safe to remove them until we've landed on an exact design for that.

**Pull request merged**